### DISCUSSION TITLE
Preloading Firefox for quicker load time

### DISCUSSION THREAD

**Discussion by Swefton** [hyprwm/Hyprland#6792] [Q&A] [answered]
I've noticed that if I have a Firefox window open, then opening a new window is far quicker than starting from scratch. I wanted to find out if there's a way to keep Firefox running in the background for this quick startup.

I tried using [Preload](https://wiki.archlinux.org/title/preload) to mixed results, it was working for a bit but then the service wouldn't start and it kept giving trouble.

I was wondering if an alternative approach could be leaving Firefox running in a hyprland tray or some other alternative?

**Comment by fufexan** [comment, marked as answer]
Firefox does not use SNI, so it can't be left in a tray. Best solution is to use Preload. Alternatively, try starting firefox in a clean profile and see what extensions and settings you can disable/configure from your usual profile to make it load faster.

**Reply by Swefton** [reply]
Minimize to system didn't work, I tried following the [Steam Solution](https://wiki.hyprland.org/Configuring/Uncommon-tips--tricks/) but firefox wouldn't close nor minimize.

```
if [ "$(hyprctl activewindow -j | jq -r ".class")" = "Steam" ]; then
    xdotool getactivewindow windowunmap
else
    hyprctl dispatch killactive ""
fi
```

I suppose I'll stick to the preload solution.
